cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,204 @@
 from housekeeper.store.models import Version
 
 from cg.apps.slurm.slurm_api import SlurmAPI
-from cg.constants import HK_FASTQ_TAGS
+from cg.constants import HK_FASTQ_TAGS, FileExtensions
+from cg.constants.constants import CaseActions
 from cg.meta.meta import MetaAPI
 from cg.meta.rsync.sbatch import ERROR_RSYNC_FUNCTION, RSYNC_CONTENTS_COMMAND
+from cg.meta.transfer.utils import are_all_fastq_valid
 from cg.models.cg_config import CGConfig
 from cg.models.slurm.sbatch import Sbatch
 from cg.store.models import Case, Customer, Sample
-from cg.utils.checksum.checksum import check_md5sum, extract_md5sum
+from cg.utils.files import get_files_matching_pattern
 
 LOG = logging.getLogger(__name__)
 
 
 class ExternalDataAPI(MetaAPI):
     def __init__(self, config: CGConfig):
         super().__init__(config)
-        self.destination_path: str = config.external.hasta
-        self.source_path: str = config.external.caesar
-        self.base_path: str = config.data_delivery.base_path
         self.account: str = config.data_delivery.account
+        self.base_path: str = config.data_delivery.base_path
+        self.customer_id: str | None = None
+        self.destination_path: str = config.external.hasta
+        self.dry_run: bool | None = None
+        self.force: bool | None = None
         self.mail_user: str = config.data_delivery.mail_user
-        self.slurm_api: SlurmAPI = SlurmAPI()
         self.RSYNC_FILE_POSTFIX: str = "_rsync_external_data"
+        self.slurm_api: SlurmAPI = SlurmAPI()
+        self.source_path: str = config.external.caesar
+        self.ticket: str | None = None
```

Comment:
We agreed that they would not be parameters for the init function. I though of defining them here so that we only define attributes here. Do you mean defining them inside the endpoint methods? @islean

Reply:
Oh yeah right! To me an API does not have hold any data apart from configuration needed for it to "boot". Any data specific for a method call such as customer_id or ticket I would pass between methods. But I do not know if that is the agreed universal way to do things.